Marlin/src/module/ft_motion.cpp

```diff
@@ -172,86 +145,12 @@ void FTMotion::loop() {
     reset();
     stepper.abort_current_block = false;  // Abort finished.
   }
+  static uint32_t was = 0;
 
-  while (!blockProcRdy && (stepper.current_block = planner.get_current_block())) {
-    if (stepper.current_block->is_sync()) {     // Sync block?
-      if (stepper.current_block->is_sync_pos()) // Position sync? Set the position.
-        stepper._set_position(stepper.current_block->position);
-      discard_planner_block_protected();
-      continue;
-    }
-    loadBlockData(stepper.current_block);
-
-    #if ENABLED(POWER_LOSS_RECOVERY)
-      recovery.info.sdpos = stepper.current_block->sdpos;
-      recovery.info.current_position = stepper.current_block->start_position;
-    #endif
-
-    blockProcRdy = true;
-
-    // Some kinematics track axis motion in HX, HY, HZ
-    #if ANY(CORE_IS_XY, CORE_IS_XZ, MARKFORGED_XY, MARKFORGED_YX)
-      stepper.last_direction_bits.hx = stepper.current_block->direction_bits.hx;
-    #endif
-    #if ANY(CORE_IS_XY, CORE_IS_YZ, MARKFORGED_XY, MARKFORGED_YX)
-      stepper.last_direction_bits.hy = stepper.current_block->direction_bits.hy;
-    #endif
-    #if ANY(CORE_IS_XZ, CORE_IS_YZ)
-      stepper.last_direction_bits.hz = stepper.current_block->direction_bits.hz;
-    #endif
-  }
-
-  if (blockProcRdy) {
-
-    if (!batchRdy) generateTrajectoryPointsFromBlock(); // may clear blockProcRdy
-
-    // Check if the block has been completely converted:
-    if (!blockProcRdy) {
-      discard_planner_block_protected();
-      if (!batchRdy && !planner.has_blocks_queued()) {
-        runoutBlock();
-        generateTrajectoryPointsFromBlock(); // Additional call to guarantee batchRdy is set this loop.
-      }
-    }
-  }
-
-  // FBS / post processing.
-  if (batchRdy && !batchRdyForInterp) {
-
-    // Call Ulendo FBS here.
```

Comment:
Ulendo themselves may need this hook for their own internal stuff.

Reply:
Our recent optimizations are also fixing critical motion shortcomings so we hope to move forward and get this thing shipped with these improvements. But this and the updated `M493` means more work for you with FBS @ulendoalex . Can you work with these changes?

Reply:
It should involve only adding your buffer between these two calls:
```cpp
    xyze_float_t traj_coords = calc_traj_point(currentGenerator->getDistanceAtTime(tau));
    traj_coords = add_to_ulendo_buffer_and_return_a_transformed_coord_point
    stepper_plan_t plan = calc_stepper_plan(traj_coords);
```
and adding an extra waiting time as big as your buffer to the runout.

You'll also benefit from all the time resolution, perf and ram improvements (including ultra clean & silent pulse trains, increased surface quality, reduced resonances, better linear advance, etc).